packages/uilib/src/lib/plugins/type-designer/components/controls/event-handlers.ts

```diff
@@ -0,0 +1,76 @@
+// event-handlers.ts
+import { EditorEventHandler } from "../../editor-events/editor-event-handler";
+import { 
+    type CreateBayEvent, 
+    type CreateSubstationEvent, 
+    type CreateLDeviceEvent, 
+    type CreateIEDEvent, 
+    type CreateVoltageLevelEvent 
+} from "../../editor-events/event-types";
+
+export function onCreateBay(event: CustomEvent<CreateBayEvent>, editEventHandler: EditorEventHandler) {
+    console.log("onCreateBay:", event.detail.type.name);
+    editEventHandler.dispatchCreateBay(event.detail);
+}
+
+export function onCreateSubstation(event: CustomEvent<CreateSubstationEvent>, editEventHandler: EditorEventHandler) {
+    console.log("onCreateSubstation:", event.detail.type.name);
+    editEventHandler.dispatchCreateSubstation(event.detail);
+}
+
+export function onCreateLDevice(event: CustomEvent<CreateLDeviceEvent>, editEventHandler: EditorEventHandler) {
+    console.log("onCreateLDevice:", event.detail.type.inst);
+    editEventHandler.dispatchCreateLDevice(event.detail);
+}
+
+export function onCreateIED(event: CustomEvent<CreateIEDEvent>, editEventHandler: EditorEventHandler) {
+    console.log("onCreateIED:", event.detail.type.name);
+    editEventHandler.dispatchCreateIED(event.detail);
+}
+
+export function onCreateVoltageLevel(event: CustomEvent<CreateVoltageLevelEvent>, editEventHandler: EditorEventHandler) {
+    console.log("onCreateVoltageLevel:", event.detail.type.name);
+    editEventHandler.dispatchCreateVoltageLevel(event.detail);
+}
+
+type EventTypes = 
+    | CreateBayEvent
+    | CreateSubstationEvent
+    | CreateLDeviceEvent
+    | CreateIEDEvent
+    | CreateVoltageLevelEvent;
+
+// my inspiration - https://stackoverflow.com/questions/60067100/why-is-the-infer-keyword-needed-in-typescript
+type EventDetail<T> = T extends { bay: infer U } ? U :
+                      T extends { substation: infer U } ? U :
+                      T extends { lDevice: infer U } ? U :
+                      T extends { ied: infer U } ? U :
+                      T extends { vLevel: infer U } ? U : never;
+
+export function handleCreateEvent<T extends EventTypes>(eventType: string, detail: EventDetail<T>, editEventHandler: EditorEventHandler) {
```

Comment:
if I follow the flow the right way, the formData becomes here detail. If it is correct, it seems that the argument and the parameter doesn't share the same type. Is it correct ?

I'm not sure to understand why we need this complex type (`EventDetail`) compared to the previous one (`FormData`).

Reply:
use of openSCD event API